include/envoy/config/config_provider.h

```diff
@@ -92,13 +147,20 @@ class ConfigProvider {
    * @return Protobuf::Message* the config proto corresponding to the Config instantiated by the
    *         provider.
    */
-  virtual const Protobuf::Message* getConfigProto() const PURE;
+  virtual const Protobuf::Message* getConfigProto() const { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
+
+  /**
+   * Returns the config protos associated with the provider.
+   * @return const ConfigProtoVector the config protos corresponding to the Config instantiated by
+   *         the provider.
+   */
+  virtual const ConfigProtoVector getConfigProtos() const { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
```

Comment:
Should this be `const ConfigProtoVector&` if by const reference? Otherwise it's a copy, and the first const in this line can be dropped.

Reply:
Good catch; I switched between a reference and a copy and ultimately chose the latter. The common case is that the implementing function will require a conversion between the type used for storage and the ConfigProtoVector return type.

Reply:
I removed the const.